// chalo-google-hybrid-login/esm5/lib/user-login/user-login.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Output } from '@angular/core';
import { FormBuilder, Validators } from '@angular/forms';
import { GoogleLoginService } from '../chalo-google-hybrid-login.service';
import { StatusCode } from '../models';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/forms';
import * as ɵngcc2 from '@angular/material/form-field';
import * as ɵngcc3 from '@angular/material/input';
var UserLoginComponent = /** @class */ (function () {
    function UserLoginComponent(fb, loginService) {
        this.fb = fb;
        this.loginService = loginService;
        this.onSignInResponse = new EventEmitter();
        this.initializeForm();
    }
    /**
     * @return {?}
     */
    UserLoginComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    UserLoginComponent.prototype.initializeForm = /**
     * @return {?}
     */
    function () {
        this.loginForm = this.fb.group({
            userId: [null, Validators.required],
            staticPassword: [null, Validators.required]
        });
    };
    /**
     * @return {?}
     */
    UserLoginComponent.prototype.onSubmit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.loginForm.valid) {
            /** @type {?} */
            var body = this.loginForm.getRawValue();
            this.loginService.validateUser({
                data: this.loginService.encryptCookie(JSON.stringify(body))
            }).subscribe(function (res) {
                _this.loginService.getUserToken(res).subscribe(function (data) {
                    _this.loginService.setCookieData(data);
                    _this.onSignInResponse.emit({ code: StatusCode.Success, data: data });
                }, function (err) {
                    _this.onSignInResponse.emit({ code: StatusCode.Failure, data: err });
                });
            }, function (err) {
                _this.onSignInResponse.emit({ code: StatusCode.Failure, data: err });
            });
        }
    };
    /** @nocollapse */
    UserLoginComponent.ctorParameters = function () { return [
        { type: FormBuilder },
        { type: GoogleLoginService }
    ]; };
    UserLoginComponent.propDecorators = {
        onSignInResponse: [{ type: Output }]
    };
UserLoginComponent.ɵfac = function UserLoginComponent_Factory(t) { return new (t || UserLoginComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormBuilder), ɵngcc0.ɵɵdirectiveInject(GoogleLoginService)); };
UserLoginComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: UserLoginComponent, selectors: [["user-login"]], outputs: { onSignInResponse: "onSignInResponse" }, decls: 10, vars: 2, consts: [[1, "form-group"], [3, "formGroup"], ["appearance", "outline"], ["matInput", "", "placeholder", "Enter User Id", "formControlName", "userId", "required", "", "autocomplete", "off"], ["matInput", "", "type", "password", "placeholder", "Enter Password", "formControlName", "staticPassword", "required", "", "autocomplete", "off"], [1, "button-container"], ["type", "button", 1, "cstm-button", "active", 3, "disabled", "click"], [1, "btn-text"]], template: function UserLoginComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "form", 1);
        ɵngcc0.ɵɵelementStart(2, "mat-form-field", 2);
        ɵngcc0.ɵɵelement(3, "input", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "mat-form-field", 2);
        ɵngcc0.ɵɵelement(5, "input", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "div", 5);
        ɵngcc0.ɵɵelementStart(7, "button", 6);
        ɵngcc0.ɵɵlistener("click", function UserLoginComponent_Template_button_click_7_listener() { return ctx.onSubmit(); });
        ɵngcc0.ɵɵelementStart(8, "a", 7);
        ɵngcc0.ɵɵtext(9, "Continue");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("formGroup", ctx.loginForm);
        ɵngcc0.ɵɵadvance(6);
        ɵngcc0.ɵɵproperty("disabled", !ctx.loginForm.valid);
    } }, directives: [ɵngcc1.ɵangular_packages_forms_forms_y, ɵngcc1.NgControlStatusGroup, ɵngcc1.FormGroupDirective, ɵngcc2.MatFormField, ɵngcc3.MatInput, ɵngcc1.DefaultValueAccessor, ɵngcc1.NgControlStatus, ɵngcc1.FormControlName, ɵngcc1.RequiredValidator], styles: [".form-group[_ngcontent-%COMP%]{text-align:center;font-size:12px}.form-control[_ngcontent-%COMP%]{height:40px;border-radius:2px;box-shadow:none}.form-control[_ngcontent-%COMP%]:focus{border-color:#eee}.cstm-button[_ngcontent-%COMP%]{background:0 0;border:1px solid #f47c21;border-radius:5px;text-align:center;width:100%;padding:8px 10px;display:inline-block;line-height:20px;margin-left:0!important}.btn-text[_ngcontent-%COMP%]{text-decoration:none;color:#f47c21;font-weight:700;font-size:14px}.cstm-button[_ngcontent-%COMP%]:disabled{cursor:not-allowed!important}.cstm-button.active[_ngcontent-%COMP%]{cursor:pointer;background:#f47c21}  .mat-form-field-flex{border:0!important}  .mat-form-field-infix{padding:8px 16px}  .mat-form-field-infix:focus{outline:#f47c21 solid 2px}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UserLoginComponent, [{
        type: Component,
        args: [{
                selector: 'user-login',
                template: "<div class=\"form-group\">\n  <form [formGroup]=\"loginForm\">\n    <mat-form-field appearance=\"outline\">\n      <input\n        matInput\n        placeholder=\"Enter User Id\"\n        formControlName=\"userId\"\n        required\n        autocomplete=\"off\"\n      />\n    </mat-form-field>\n    <mat-form-field appearance=\"outline\">\n      <input\n        matInput\n        type=\"password\"\n        placeholder=\"Enter Password\"\n        formControlName=\"staticPassword\"\n        required\n        autocomplete=\"off\"\n      />\n    </mat-form-field>\n    <div class=\"button-container\">\n      <button\n        class=\"cstm-button active\"\n        type=\"button\"\n        [disabled]=\"!loginForm.valid\"\n        (click)=\"onSubmit()\"\n      >\n        <a class=\"btn-text\">Continue</a>\n      </button>\n    </div>\n  </form>\n</div>\n",
                styles: [".form-group{text-align:center;font-size:12px}.form-control{height:40px;border-radius:2px;box-shadow:none}.form-control:focus{border-color:#eee}.cstm-button{background:0 0;border:1px solid #f47c21;border-radius:5px;text-align:center;width:100%;padding:8px 10px;display:inline-block;line-height:20px;margin-left:0!important}.btn-text{text-decoration:none;color:#f47c21;font-weight:700;font-size:14px}.cstm-button:disabled{cursor:not-allowed!important}.cstm-button.active{cursor:pointer;background:#f47c21}::ng-deep .mat-form-field-flex{border:0!important}::ng-deep .mat-form-field-infix{padding:8px 16px}::ng-deep .mat-form-field-infix:focus{outline:#f47c21 solid 2px}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormBuilder }, { type: GoogleLoginService }]; }, { onSignInResponse: [{
            type: Output
        }] }); })();
    return UserLoginComponent;
}());
export { UserLoginComponent };
if (false) {
    /** @type {?} */
    UserLoginComponent.prototype.loginForm;
    /** @type {?} */
    UserLoginComponent.prototype.onSignInResponse;
    /** @type {?} */
    UserLoginComponent.prototype.fb;
    /** @type {?} */
    UserLoginComponent.prototype.loginService;
}
