// chalo-google-hybrid-login/esm5/lib/models/chalo-google-login-service.interceptor.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { InjectionToken, Optional, Inject, Injectable } from "@angular/core";
import { CookieService } from "ngx-cookie-service";
import { GoogleLoginService } from "../chalo-google-hybrid-login.service";
import { GOOGLE_LOGIN_CONFIG } from "./chalo-google-login.config";
/** @type {?} */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'ngx-cookie-service';
export var GOOGLE_LOGIN_HTTP_INTERCEPTOR = new InjectionToken('chalo-google-login.interceptor');
var GoogleAuthHttpInterceptor = /** @class */ (function () {
    function GoogleAuthHttpInterceptor(googleLoginInterceptor, cookieService, googleLoginService, config) {
        this.googleLoginInterceptor = googleLoginInterceptor;
        this.cookieService = cookieService;
        this.googleLoginService = googleLoginService;
        this.config = config;
    }
    /**
     * @param {?} req
     * @param {?} next
     * @return {?}
     */
    GoogleAuthHttpInterceptor.prototype.intercept = /**
     * @param {?} req
     * @param {?} next
     * @return {?}
     */
    function (req, next) {
        if (!(req.url.includes('token') || req.url.includes('authenticated') || req.url.includes('Otp') || req.url.includes('logout')) && this.cookieService.check(this.config.user.loginStateCookie)) {
            /** @type {?} */
            var loginCookie = this.cookieService.get(this.config.user.loginStateCookie);
            if (loginCookie) {
                /** @type {?} */
                var decryptLoginCookie = this.googleLoginService.decryptString(loginCookie);
                if (decryptLoginCookie && decryptLoginCookie === 'loggedOut') {
                    this.googleLoginService.sendLogoutMessage('Signed out');
                }
            }
        }
        if (this.googleLoginInterceptor) {
            return this.googleLoginInterceptor.intercept(req, next);
        }
        else {
            return next.handle(req);
        }
    };
    /** @nocollapse */
    GoogleAuthHttpInterceptor.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [GOOGLE_LOGIN_HTTP_INTERCEPTOR,] }] },
        { type: CookieService },
        { type: GoogleLoginService },
        { type: undefined, decorators: [{ type: Inject, args: [GOOGLE_LOGIN_CONFIG,] }] }
    ]; };
GoogleAuthHttpInterceptor.ɵfac = function GoogleAuthHttpInterceptor_Factory(t) { return new (t || GoogleAuthHttpInterceptor)(ɵngcc0.ɵɵinject(GOOGLE_LOGIN_HTTP_INTERCEPTOR, 8), ɵngcc0.ɵɵinject(ɵngcc1.CookieService), ɵngcc0.ɵɵinject(GoogleLoginService), ɵngcc0.ɵɵinject(GOOGLE_LOGIN_CONFIG)); };
GoogleAuthHttpInterceptor.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: GoogleAuthHttpInterceptor, factory: function (t) { return GoogleAuthHttpInterceptor.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GoogleAuthHttpInterceptor, [{
        type: Injectable
    }], function () { return [{ type: undefined, decorators: [{
                type: Optional
            }, {
                type: Inject,
                args: [GOOGLE_LOGIN_HTTP_INTERCEPTOR]
            }] }, { type: ɵngcc1.CookieService }, { type: GoogleLoginService }, { type: undefined, decorators: [{
                type: Inject,
                args: [GOOGLE_LOGIN_CONFIG]
            }] }]; }, null); })();
    return GoogleAuthHttpInterceptor;
}());
export { GoogleAuthHttpInterceptor };
if (false) {
    /** @type {?} */
    GoogleAuthHttpInterceptor.prototype.googleLoginInterceptor;
    /** @type {?} */
    GoogleAuthHttpInterceptor.prototype.cookieService;
    /** @type {?} */
    GoogleAuthHttpInterceptor.prototype.googleLoginService;
    /** @type {?} */
    GoogleAuthHttpInterceptor.prototype.config;
}
