// chalo-google-hybrid-login/esm5/lib/directives/number.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Directive, ElementRef, HostListener } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var NumberDirective = /** @class */ (function () {
    function NumberDirective(el) {
        this.el = el;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    NumberDirective.prototype.onPaste = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        event.preventDefault();
        /** @type {?} */
        var pastedInput = event.clipboardData
            .getData('text/plain')
            .replace(/\D/g, ''); // get a digit-only string
        document.execCommand('insertText', false, pastedInput);
    };
    /**
     * @param {?} e
     * @return {?}
     */
    NumberDirective.prototype.onKeyDown = /**
     * @param {?} e
     * @return {?}
     */
    function (e) {
        if (
        // Allow: Delete, Backspace, Tab, Escape, Enter
        [46, 8, 9, 27, 13].indexOf(e.keyCode) !== -1 ||
            (e.keyCode === 65 && e.ctrlKey === true) || // Allow: Ctrl+A
            // Allow: Ctrl+A
            (e.keyCode === 67 && e.ctrlKey === true) || // Allow: Ctrl+C
            // Allow: Ctrl+C
            (e.keyCode === 86 && e.ctrlKey === true) || // Allow: Ctrl+V
            // Allow: Ctrl+V
            (e.keyCode === 88 && e.ctrlKey === true) || // Allow: Ctrl+X
            // Allow: Ctrl+X
            (e.keyCode === 65 && e.metaKey === true) || // Cmd+A (Mac)
            // Cmd+A (Mac)
            (e.keyCode === 67 && e.metaKey === true) || // Cmd+C (Mac)
            // Cmd+C (Mac)
            (e.keyCode === 86 && e.metaKey === true) || // Cmd+V (Mac)
            // Cmd+V (Mac)
            (e.keyCode === 88 && e.metaKey === true) || // Cmd+X (Mac)
            // Cmd+X (Mac)
            (e.keyCode >= 35 && e.keyCode <= 39) // Home, End, Left, Right
        ) {
            return; // let it happen, don't do anything
        }
        // Ensure that it is a number and stop the keypress
        if ((e.shiftKey || (e.keyCode < 48 || e.keyCode > 57)) &&
            (e.keyCode < 96 || e.keyCode > 105)) {
            e.preventDefault();
        }
    };
    /** @nocollapse */
    NumberDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    NumberDirective.propDecorators = {
        onPaste: [{ type: HostListener, args: ['paste', ['$event'],] }],
        onKeyDown: [{ type: HostListener, args: ['keydown', ['$event'],] }]
    };
NumberDirective.ɵfac = function NumberDirective_Factory(t) { return new (t || NumberDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
NumberDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NumberDirective, selectors: [["", "numberDirective", ""]], hostBindings: function NumberDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("paste", function NumberDirective_paste_HostBindingHandler($event) { return ctx.onPaste($event); })("keydown", function NumberDirective_keydown_HostBindingHandler($event) { return ctx.onKeyDown($event); });
    } } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NumberDirective, [{
        type: Directive,
        args: [{
                selector: '[numberDirective]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { onPaste: [{
            type: HostListener,
            args: ['paste', ['$event']]
        }], onKeyDown: [{
            type: HostListener,
            args: ['keydown', ['$event']]
        }] }); })();
    return NumberDirective;
}());
export { NumberDirective };
if (false) {
    /** @type {?} */
    NumberDirective.prototype.el;
}
