// generic-list-filter/esm2015/lib/generic-list-filter.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgxMaterialTimepickerModule } from 'ngx-material-timepicker';
import { GenericListFilterComponent } from './generic-list-filter.component';
import * as ɵngcc0 from '@angular/core';
export class GenericListFilterModule {
}
GenericListFilterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: GenericListFilterModule });
GenericListFilterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function GenericListFilterModule_Factory(t) { return new (t || GenericListFilterModule)(); }, imports: [[
            CommonModule,
            ReactiveFormsModule,
            FormsModule,
            NgxMaterialTimepickerModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(GenericListFilterModule, { declarations: function () { return [GenericListFilterComponent]; }, imports: function () { return [CommonModule,
        ReactiveFormsModule,
        FormsModule,
        NgxMaterialTimepickerModule]; }, exports: function () { return [GenericListFilterComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GenericListFilterModule, [{
        type: NgModule,
        args: [{
                declarations: [GenericListFilterComponent],
                imports: [
                    CommonModule,
                    ReactiveFormsModule,
                    FormsModule,
                    NgxMaterialTimepickerModule
                ],
                exports: [GenericListFilterComponent]
            }]
    }], null, null); })();
