// chalo-google-hybrid-login/esm2015/lib/script-load.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { Observable } from 'rxjs';
import { take, shareReplay } from 'rxjs/operators';
import { DOCUMENT } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
export class ScriptLoadService {
    /**
     * @param {?} document
     */
    constructor(document) {
        this.document = document;
        this.scriptsLoaders = {};
    }
    /**
     *
     * @param {?} url Url of the external script to be loaded
     * @param {?=} attributes Attribute list to be added to the script element
     * @param {?=} targetEl Target element for the placing script tag. It can be a selector or a element reference
     * @return {?}
     */
    loadScript(url, attributes, targetEl = 'head') {
        return this.scriptsLoaders[url] = this.scriptsLoaders[url] || new Observable((observer) => {
            /** @type {?} */
            const script = this.document.createElement('script');
            if (attributes) {
                for (const key in attributes) {
                    if (attributes.hasOwnProperty(key)) {
                        script.setAttribute(key, attributes[key]);
                    }
                }
            }
            script.onload = (event) => {
                console.log('onload', event);
                observer.next(event);
                observer.complete();
            };
            script.onerror = err => {
                console.log('sdad', err);
                observer.error(err);
            };
            script.src = url;
            /** @type {?} */
            const targetElement = typeof targetEl === 'string' ? this.document.querySelector(targetEl) : targetEl;
            targetElement.appendChild(script);
        })
            .pipe(take(1), shareReplay(1));
    }
}
ScriptLoadService.ɵfac = function ScriptLoadService_Factory(t) { return new (t || ScriptLoadService)(ɵngcc0.ɵɵinject(DOCUMENT)); };
ScriptLoadService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: ScriptLoadService, factory: ScriptLoadService.ɵfac });
/** @nocollapse */
ScriptLoadService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ScriptLoadService, [{
        type: Injectable
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [DOCUMENT]
            }] }]; }, null); })();
if (false) {
    /** @type {?} */
    ScriptLoadService.prototype.scriptsLoaders;
    /** @type {?} */
    ScriptLoadService.prototype.document;
}
