// chalo-google-hybrid-login/esm5/lib/chalo-google-hybrid-login.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { ChaloGoogleHybridLoginComponent } from './chalo-google-hybrid-login.component';
import { HttpClientModule } from "@angular/common/http";
import { GoogleLoginService } from "./chalo-google-hybrid-login.service";
import { HTTP_INTERCEPTORS } from "@angular/common/http";
import { GoogleAuthHttpInterceptor } from "./models/chalo-google-login-service.interceptor";
import { ScriptLoadService } from './script-load.service';
import { CommonModule } from '@angular/common';
import { CookieService } from 'ngx-cookie-service';
import { MobieLoginComponent } from './mobie-login/mobie-login.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatInputModule } from '@angular/material/input';
import { MatFormFieldModule } from '@angular/material/form-field';
import { NumberDirective } from './directives/number.directive';
import { NgOtpInputModule } from 'ng-otp-input';
import { UserLoginComponent } from './user-login/user-login.component';
import * as ɵngcc0 from '@angular/core';
var ChaloGoogleHybridLoginModule = /** @class */ (function () {
    function ChaloGoogleHybridLoginModule() {
    }
ChaloGoogleHybridLoginModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ChaloGoogleHybridLoginModule });
ChaloGoogleHybridLoginModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ChaloGoogleHybridLoginModule_Factory(t) { return new (t || ChaloGoogleHybridLoginModule)(); }, providers: [GoogleLoginService, ScriptLoadService, CookieService,
        { provide: HTTP_INTERCEPTORS, useClass: GoogleAuthHttpInterceptor, multi: true }], imports: [[HttpClientModule, CommonModule, FormsModule, ReactiveFormsModule,
            MatFormFieldModule, MatInputModule, NgOtpInputModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ChaloGoogleHybridLoginModule, { declarations: function () { return [ChaloGoogleHybridLoginComponent, MobieLoginComponent, NumberDirective, UserLoginComponent]; }, imports: function () { return [HttpClientModule, CommonModule, FormsModule, ReactiveFormsModule,
        MatFormFieldModule, MatInputModule, NgOtpInputModule]; }, exports: function () { return [ChaloGoogleHybridLoginComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChaloGoogleHybridLoginModule, [{
        type: NgModule,
        args: [{
                imports: [HttpClientModule, CommonModule, FormsModule, ReactiveFormsModule,
                    MatFormFieldModule, MatInputModule, NgOtpInputModule
                ],
                declarations: [ChaloGoogleHybridLoginComponent, MobieLoginComponent, NumberDirective, UserLoginComponent],
                exports: [ChaloGoogleHybridLoginComponent],
                providers: [GoogleLoginService, ScriptLoadService, CookieService,
                    { provide: HTTP_INTERCEPTORS, useClass: GoogleAuthHttpInterceptor, multi: true }]
            }]
    }], function () { return []; }, null); })();
    return ChaloGoogleHybridLoginModule;
}());
export { ChaloGoogleHybridLoginModule };
