// chalo-google-hybrid-login/esm5/lib/script-load.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import { Observable } from 'rxjs';
import { take, shareReplay } from 'rxjs/operators';
import { DOCUMENT } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
var ScriptLoadService = /** @class */ (function () {
    function ScriptLoadService(document) {
        this.document = document;
        this.scriptsLoaders = {};
    }
    /**
     *
     * @param url Url of the external script to be loaded
     * @param attributes Attribute list to be added to the script element
     * @param targetEl Target element for the placing script tag. It can be a selector or a element reference
     */
    /**
     *
     * @param {?} url Url of the external script to be loaded
     * @param {?=} attributes Attribute list to be added to the script element
     * @param {?=} targetEl Target element for the placing script tag. It can be a selector or a element reference
     * @return {?}
     */
    ScriptLoadService.prototype.loadScript = /**
     *
     * @param {?} url Url of the external script to be loaded
     * @param {?=} attributes Attribute list to be added to the script element
     * @param {?=} targetEl Target element for the placing script tag. It can be a selector or a element reference
     * @return {?}
     */
    function (url, attributes, targetEl) {
        var _this = this;
        if (targetEl === void 0) { targetEl = 'head'; }
        return this.scriptsLoaders[url] = this.scriptsLoaders[url] || new Observable(function (observer) {
            /** @type {?} */
            var script = _this.document.createElement('script');
            if (attributes) {
                for (var key in attributes) {
                    if (attributes.hasOwnProperty(key)) {
                        script.setAttribute(key, attributes[key]);
                    }
                }
            }
            script.onload = function (event) {
                console.log('onload', event);
                observer.next(event);
                observer.complete();
            };
            script.onerror = function (err) {
                console.log('sdad', err);
                observer.error(err);
            };
            script.src = url;
            /** @type {?} */
            var targetElement = typeof targetEl === 'string' ? _this.document.querySelector(targetEl) : targetEl;
            targetElement.appendChild(script);
        })
            .pipe(take(1), shareReplay(1));
    };
    /** @nocollapse */
    ScriptLoadService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
    ]; };
ScriptLoadService.ɵfac = function ScriptLoadService_Factory(t) { return new (t || ScriptLoadService)(ɵngcc0.ɵɵinject(DOCUMENT)); };
ScriptLoadService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: ScriptLoadService, factory: function (t) { return ScriptLoadService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ScriptLoadService, [{
        type: Injectable
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [DOCUMENT]
            }] }]; }, null); })();
    return ScriptLoadService;
}());
export { ScriptLoadService };
if (false) {
    /** @type {?} */
    ScriptLoadService.prototype.scriptsLoaders;
    /** @type {?} */
    ScriptLoadService.prototype.document;
}
