// generic-list-filter/__ivy_ngcc__/fesm2015/generic-list-filter.js
import { EventEmitter, Component, Input, Output, NgModule } from '@angular/core';
import * as moment from 'moment';
import { findIndex } from 'lodash';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';
import { NgxMaterialTimepickerModule } from 'ngx-material-timepicker';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/forms';
import * as ɵngcc3 from 'ngx-material-timepicker';

function GenericListFilterComponent_div_2_ng_container_1_div_1_Template(rf, ctx) { if (rf & 1) {
    const _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 11);
    ɵngcc0.ɵɵelementStart(1, "span", 12);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "titlecase");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "span", 13);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_2_ng_container_1_div_1_Template_span_click_4_listener() { ɵngcc0.ɵɵrestoreView(_r10); const item_r7 = ctx.$implicit; const ctx_r9 = ɵngcc0.ɵɵnextContext(3); return ctx_r9.onRemoveFilter(item_r7); });
    ɵngcc0.ɵɵtext(5, " x ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r7 = ctx.$implicit;
    const ctx_r6 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate2(" ", ɵngcc0.ɵɵpipeBind1(3, 2, item_r7), " : ", ctx_r6.allAppliedFilters[item_r7], " ");
} }
function GenericListFilterComponent_div_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, GenericListFilterComponent_div_2_ng_container_1_div_1_Template, 6, 4, "div", 10);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r2.filterKeys);
} }
function GenericListFilterComponent_div_2_div_2_Template(rf, ctx) { if (rf & 1) {
    const _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 11);
    ɵngcc0.ɵɵelementStart(1, "span", 12);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "titlecase");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "span", 13);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_2_div_2_Template_span_click_4_listener() { ɵngcc0.ɵɵrestoreView(_r12); const ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.onRemoveFilter(ctx_r11.filterKeys[0]); });
    ɵngcc0.ɵɵtext(5, " x ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate2(" ", ɵngcc0.ɵɵpipeBind1(3, 2, ctx_r3.filterKeys[0]), " : ", ctx_r3.allAppliedFilters[ctx_r3.filterKeys[0]], " ");
} }
function GenericListFilterComponent_div_2_div_3_Template(rf, ctx) { if (rf & 1) {
    const _r14 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 11);
    ɵngcc0.ɵɵelementStart(1, "span", 12);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵpipe(3, "titlecase");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "span", 13);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_2_div_3_Template_span_click_4_listener() { ɵngcc0.ɵɵrestoreView(_r14); const ctx_r13 = ɵngcc0.ɵɵnextContext(2); return ctx_r13.onRemoveFilter(ctx_r13.filterKeys[1]); });
    ɵngcc0.ɵɵtext(5, " x ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate2(" ", ɵngcc0.ɵɵpipeBind1(3, 2, ctx_r4.filterKeys[1]), " : ", ctx_r4.allAppliedFilters[ctx_r4.filterKeys[1]], " ");
} }
function GenericListFilterComponent_div_2_div_4_Template(rf, ctx) { if (rf & 1) {
    const _r16 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 14);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_2_div_4_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r16); const ctx_r15 = ɵngcc0.ɵɵnextContext(2); return ctx_r15.openRestAppliedFilters(); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r5 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" +", ctx_r5.filterKeys.length - 2, " ");
} }
function GenericListFilterComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtemplate(1, GenericListFilterComponent_div_2_ng_container_1_Template, 2, 1, "ng-container", 7);
    ɵngcc0.ɵɵtemplate(2, GenericListFilterComponent_div_2_div_2_Template, 6, 4, "div", 8);
    ɵngcc0.ɵɵtemplate(3, GenericListFilterComponent_div_2_div_3_Template, 6, 4, "div", 8);
    ɵngcc0.ɵɵtemplate(4, GenericListFilterComponent_div_2_div_4_Template, 2, 1, "div", 9);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ctx_r0.filterKeys.length > 2 ? "filter-list-container-2 flex-wrap" : "filter-list-container-2");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.isCountClicked);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.isCountClicked);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (ctx_r0.filterKeys == null ? null : ctx_r0.filterKeys.length) >= 2 && !ctx_r0.isCountClicked);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.filterKeys.length > 2 && !ctx_r0.isCountClicked);
} }
function GenericListFilterComponent_div_5_div_15_Template(rf, ctx) { if (rf & 1) {
    const _r25 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 33);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_5_div_15_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r25); const filter_r23 = ctx.$implicit; const ctx_r24 = ɵngcc0.ɵɵnextContext(2); return ctx_r24.selectedFilterTab(filter_r23); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "titlecase");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const filter_r23 = ctx.$implicit;
    const ctx_r17 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassProp("selected", ctx_r17.selectedFilterTabType1 === filter_r23 || ctx_r17.isFilterExists(filter_r23));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind1(2, 3, filter_r23), " ");
} }
function GenericListFilterComponent_div_5_div_16_Template(rf, ctx) { if (rf & 1) {
    const _r27 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 33);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_5_div_16_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r27); const ctx_r26 = ɵngcc0.ɵɵnextContext(2); return ctx_r26.selectedFilterTab("time"); });
    ɵngcc0.ɵɵtext(1, " Select Time ");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r18 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassProp("selected", ctx_r18.selectedFilterTabType1 === "time" || ctx_r18.timeFilter);
} }
function GenericListFilterComponent_div_5_div_19_Template(rf, ctx) { if (rf & 1) {
    const _r29 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 34);
    ɵngcc0.ɵɵelementStart(1, "input", 35);
    ɵngcc0.ɵɵlistener("ngModelChange", function GenericListFilterComponent_div_5_div_19_Template_input_ngModelChange_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r29); const ctx_r28 = ɵngcc0.ɵɵnextContext(2); return ctx_r28.searchKey = $event; })("keyup", function GenericListFilterComponent_div_5_div_19_Template_input_keyup_1_listener() { ɵngcc0.ɵɵrestoreView(_r29); const ctx_r30 = ɵngcc0.ɵɵnextContext(2); return ctx_r30.filterData(ctx_r30.searchKey); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r19 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate1("placeholder", "Search ", ctx_r19.selectedFilterTabType1, "");
    ɵngcc0.ɵɵproperty("ngModel", ctx_r19.searchKey);
} }
function GenericListFilterComponent_div_5_div_21_Template(rf, ctx) { if (rf & 1) {
    const _r33 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 36);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_5_div_21_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r33); const item_r31 = ctx.$implicit; const ctx_r32 = ɵngcc0.ɵɵnextContext(2); return ctx_r32.selectFilter(item_r31, ctx_r32.selectedFilterTabType1); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r31 = ctx.$implicit;
    const ctx_r20 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassProp("selected", ctx_r20.selectedCategory.value === item_r31);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", item_r31, " ");
} }
function GenericListFilterComponent_div_5_div_22_div_1_Template(rf, ctx) { if (rf & 1) {
    const _r37 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 36);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_5_div_22_div_1_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r37); const item_r35 = ctx.$implicit; const ctx_r36 = ɵngcc0.ɵɵnextContext(3); return ctx_r36.selectFilter(item_r35, "live status"); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r35 = ctx.$implicit;
    const ctx_r34 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵclassProp("selected", ctx_r34.selectedCategory.value.value === item_r35.value);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", item_r35.name, " ");
} }
function GenericListFilterComponent_div_5_div_22_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtemplate(1, GenericListFilterComponent_div_5_div_22_div_1_Template, 2, 3, "div", 30);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r21 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngClass", (ctx_r21.liveStatus == null ? null : ctx_r21.liveStatus.length) > 6 ? "filter-data-list-scroll-2 filter-data-list" : "filter-data-list");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r21.liveStatus);
} }
function GenericListFilterComponent_div_5_ng_container_23_div_1_Template(rf, ctx) { if (rf & 1) {
    const _r42 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 38);
    ɵngcc0.ɵɵelementStart(1, "div", 39);
    ɵngcc0.ɵɵelementStart(2, "label");
    ɵngcc0.ɵɵtext(3, "From Time");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "input", 40);
    ɵngcc0.ɵɵlistener("ngModelChange", function GenericListFilterComponent_div_5_ng_container_23_div_1_Template_input_ngModelChange_4_listener($event) { ɵngcc0.ɵɵrestoreView(_r42); const ctx_r41 = ɵngcc0.ɵɵnextContext(3); return ctx_r41.fromTime = $event; });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "ngx-material-timepicker", 41, 42);
    ɵngcc0.ɵɵlistener("timeSet", function GenericListFilterComponent_div_5_ng_container_23_div_1_Template_ngx_material_timepicker_timeSet_5_listener($event) { ɵngcc0.ɵɵrestoreView(_r42); const ctx_r43 = ɵngcc0.ɵɵnextContext(3); return ctx_r43.setTime("fromTime", $event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(7, "div", 39);
    ɵngcc0.ɵɵelementStart(8, "label");
    ɵngcc0.ɵɵtext(9, "To Time");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(10, "input", 43);
    ɵngcc0.ɵɵlistener("ngModelChange", function GenericListFilterComponent_div_5_ng_container_23_div_1_Template_input_ngModelChange_10_listener($event) { ɵngcc0.ɵɵrestoreView(_r42); const ctx_r44 = ɵngcc0.ɵɵnextContext(3); return ctx_r44.toTime = $event; });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(11, "ngx-material-timepicker", 41, 44);
    ɵngcc0.ɵɵlistener("timeSet", function GenericListFilterComponent_div_5_ng_container_23_div_1_Template_ngx_material_timepicker_timeSet_11_listener($event) { ɵngcc0.ɵɵrestoreView(_r42); const ctx_r45 = ɵngcc0.ɵɵnextContext(3); return ctx_r45.setTime("toTime", $event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const _r39 = ɵngcc0.ɵɵreference(6);
    const _r40 = ɵngcc0.ɵɵreference(12);
    const ctx_r38 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵproperty("ngModel", ctx_r38.fromTime)("ngxTimepicker", _r39);
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵproperty("ngModel", ctx_r38.toTime)("ngxTimepicker", _r40);
} }
function GenericListFilterComponent_div_5_ng_container_23_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, GenericListFilterComponent_div_5_ng_container_23_div_1_Template, 13, 4, "div", 37);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r22 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r22.selectedFilterTabType === "time");
} }
function GenericListFilterComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    const _r47 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 15);
    ɵngcc0.ɵɵelementStart(1, "div", 16);
    ɵngcc0.ɵɵelementStart(2, "div");
    ɵngcc0.ɵɵelementStart(3, "div", 17);
    ɵngcc0.ɵɵelementStart(4, "div");
    ɵngcc0.ɵɵtext(5, " Filter ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "div", 18);
    ɵngcc0.ɵɵelementStart(7, "span", 19);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_5_Template_span_click_7_listener() { ɵngcc0.ɵɵrestoreView(_r47); const ctx_r46 = ɵngcc0.ɵɵnextContext(); return ctx_r46.resetAllFilters(); });
    ɵngcc0.ɵɵtext(8, "Reset");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(9, "span", 20);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_5_Template_span_click_9_listener() { ɵngcc0.ɵɵrestoreView(_r47); const ctx_r48 = ɵngcc0.ɵɵnextContext(); return ctx_r48.onCancel(); });
    ɵngcc0.ɵɵelement(10, "img", 21);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(11, "hr");
    ɵngcc0.ɵɵelementStart(12, "div", 22);
    ɵngcc0.ɵɵelementStart(13, "div", 23);
    ɵngcc0.ɵɵelementStart(14, "div", 24);
    ɵngcc0.ɵɵtemplate(15, GenericListFilterComponent_div_5_div_15_Template, 3, 5, "div", 25);
    ɵngcc0.ɵɵtemplate(16, GenericListFilterComponent_div_5_div_16_Template, 2, 2, "div", 26);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(17, "div", 27);
    ɵngcc0.ɵɵelementStart(18, "div", 28);
    ɵngcc0.ɵɵtemplate(19, GenericListFilterComponent_div_5_div_19_Template, 2, 2, "div", 29);
    ɵngcc0.ɵɵelementStart(20, "div", 6);
    ɵngcc0.ɵɵtemplate(21, GenericListFilterComponent_div_5_div_21_Template, 2, 3, "div", 30);
    ɵngcc0.ɵɵtemplate(22, GenericListFilterComponent_div_5_div_22_Template, 2, 2, "div", 2);
    ɵngcc0.ɵɵtemplate(23, GenericListFilterComponent_div_5_ng_container_23_Template, 2, 1, "ng-container", 7);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(24, "div", 31);
    ɵngcc0.ɵɵelementStart(25, "button", 32);
    ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_div_5_Template_button_click_25_listener() { ɵngcc0.ɵɵrestoreView(_r47); const ctx_r49 = ɵngcc0.ɵɵnextContext(); return ctx_r49.applyFilters(); });
    ɵngcc0.ɵɵelementStart(26, "span");
    ɵngcc0.ɵɵtext(27, "Apply");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(15);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.filterKeys1);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (ctx_r1.paramList == null ? null : ctx_r1.paramList.fromTime) && (ctx_r1.paramList == null ? null : ctx_r1.paramList.toTime));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.selectedFilterTabType1 !== "time");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", (ctx_r1.list[ctx_r1.selectedFilterTabType1] == null ? null : ctx_r1.list[ctx_r1.selectedFilterTabType1].length) > 6 ? "filter-data-list-scroll-2 filter-data-list" : "filter-data-list");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.list[ctx_r1.selectedFilterTabType1]);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.paramList.selectedTab > 0 && ctx_r1.selectedFilterTabType1 == "live status");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (ctx_r1.paramList == null ? null : ctx_r1.paramList.fromTime) && (ctx_r1.paramList == null ? null : ctx_r1.paramList.toTime) && ctx_r1.selectedFilterTabType1 === "time");
} }
class GenericListFilterComponent {
    constructor() {
        this.onFilterChange = new EventEmitter();
        this.selectedValues = [];
        this.selectedCategory = {
            type: '',
            value: ''
        };
        this.isFiltersApplied = false;
        this.openFilterBar = false;
        this.timeFilter = false;
        this.searchKey = '';
        this.isCountClicked = false;
        this.isData = false;
        this.isStatus = false;
    }
    ngOnInit() {
    }
    ngOnChanges() {
        //Called before any other lifecycle hook. Use it to inject dependencies, but avoid any serious work here.
        //Add '${implements OnChanges}' to the class.
        setTimeout(() => {
            this.filterKeys1 = Object.keys(this.list);
            this.selectedFilterTabType1 = this.filterKeys1[0];
            this.fromTime = this.paramList.fromTime;
            this.toTime = this.paramList.toTime;
            this.listCopy = JSON.parse(JSON.stringify(this.list));
            this.isStatus = this.list.isStatus;
            this.category = this.list.category;
            this.isData = true;
        }, 2000);
    }
    openFilterSideBar() {
        this.searchKey = '';
        this.list[this.selectedFilterTabType1] = this.listCopy[this.selectedFilterTabType1];
        this.openFilterBar = !this.openFilterBar;
    }
    onCancel() {
        this.openFilterBar = false;
    }
    onRemoveFilter(item) {
        if (item == 'time') {
            this.fromTime = moment().subtract(3, 'hours').format('LT');
            this.toTime = moment().format('LT');
            this.timeFilter = false;
        }
        delete this.allAppliedFilters[item];
        this.selectedValues = this.selectedValues.filter(obj => {
            return obj.type !== item;
        });
        console.log(item, this.allAppliedFilters, this.selectedValues);
        this.applyFilters();
    }
    applyFilters() {
        this.isFiltersApplied = true;
        this.openFilterBar = false;
        let allAppliedFilters = {};
        this.selectedValues.forEach(item => {
            allAppliedFilters[item.type] = item.value;
        });
        if (this.timeFilter) {
            allAppliedFilters['time'] = `${this.fromTime} to ${this.toTime}`;
        }
        console.log(allAppliedFilters);
        this.allAppliedFilters = allAppliedFilters;
        this.filterKeys = Object.keys(this.allAppliedFilters);
        this.filterChange();
    }
    setTime(type, event) {
        this.timeFilter = true;
        if (type === "fromTime") {
            this.fromTime = event;
        }
        else {
            this.toTime = event;
        }
        ;
    }
    filterChange() {
        let filters = {
            isFilter: true,
            appliedFilterValues: this.allAppliedFilters,
            fromTime: this.timeFilter ? this.fromTime : '',
            toTime: this.timeFilter ? this.toTime : ''
        };
        console.log(filters);
        this.onFilterChange && this.onFilterChange.emit(filters);
    }
    openRestAppliedFilters() {
        this.isCountClicked = true;
    }
    selectedFilterTab(type) {
        this.searchKey = '';
        let indx = findIndex(this.selectedValues, (item) => {
            return item.type == type;
        });
        if (indx > -1) {
            this.selectedCategory = { type: type, value: this.selectedValues[indx].value };
            this.selectFilter(this.selectedCategory.value, this.selectedCategory.type);
        }
        this.selectedFilterTabType1 = type;
    }
    isFilterExists(filter) {
        let indx = findIndex(this.selectedValues, (item) => {
            return item.type == filter && item.value;
        });
        if (indx > -1) {
            return true;
        }
    }
    selectFilter(item, type) {
        let value = {
            type: type,
            value: item
        };
        let indx = findIndex(this.selectedValues, (item) => {
            return item.type == type;
        });
        if (indx > -1) {
            this.selectedValues[indx].value = item;
            this.selectedCategory = { type: type, value: this.selectedValues[indx].value };
        }
        else {
            this.selectedValues.push(value);
            this.selectedCategory = { type: type, value: item };
        }
        console.log(this.selectedValues, this.selectedCategory);
    }
    resetAllFilters() {
        this.fromTime = moment().subtract(3, 'hours').format('LT');
        this.toTime = moment().format('LT');
        this.isFiltersApplied = false;
        this.timeFilter = false;
        this.openFilterBar = false;
        this.searchKey = '';
        this.filterKeys = '';
        this.allAppliedFilters = {};
        this.selectedValues = [];
        this.selectedFilterTabType1 = this.filterKeys1[0];
        this.selectedCategory = {
            type: '',
            value: ''
        };
        this.isCountClicked = false;
        this.filterChange();
    }
    filterData(searchValue) {
        this.list[this.selectedFilterTabType1] = this.listCopy[this.selectedFilterTabType1];
        let filterArr = this.list[this.selectedFilterTabType1];
        if (filterArr && filterArr.length) {
            const data = filterArr.filter((item) => {
                return (item.toLowerCase().startsWith(searchValue.trim().toLowerCase()));
            });
            console.log(this.searchKey, searchValue, "data=>" + data, this.list[this.selectedFilterTabType1]);
            this.list[this.selectedFilterTabType1] = [...data];
        }
    }
}
GenericListFilterComponent.ɵfac = function GenericListFilterComponent_Factory(t) { return new (t || GenericListFilterComponent)(); };
GenericListFilterComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: GenericListFilterComponent, selectors: [["app-filter"]], inputs: { list: "list", paramList: "paramList" }, outputs: { onFilterChange: "onFilterChange" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 6, vars: 2, consts: [[1, "dispatch-filters", "display-flex"], [1, "display-flex"], [3, "ngClass", 4, "ngIf"], [1, "filter-color", 3, "click"], [1, "fa", "fa-filter", "fa-2x", "icon-margin"], ["class", "filter-position-modal", 4, "ngIf"], [3, "ngClass"], [4, "ngIf"], ["class", "filter-list", 4, "ngIf"], ["class", "filter-list cr-ptr", 3, "click", 4, "ngIf"], ["class", "filter-list", 4, "ngFor", "ngForOf"], [1, "filter-list"], [1, "filter-list-data"], [1, "cancle-icon", 3, "click"], [1, "filter-list", "cr-ptr", 3, "click"], [1, "filter-position-modal"], [1, "modalbody", "ele"], [1, "header-item"], [1, "cr-ptr"], [1, "reset-filter", 3, "click"], [3, "click"], ["src", "./../../assets/images/crossIcon.png", 1, "cancle-icon-1"], [1, "main-body"], [1, "side-bar-wrapper"], [1, "filter-side-menu"], ["class", "side-menu-tabs", 3, "selected", "click", 4, "ngFor", "ngForOf"], ["class", "side-menu-tabs", 3, "selected", "click", 4, "ngIf"], [1, "vl"], [1, "filter-list-wrapper"], ["style", "margin-left: 22px;", 4, "ngIf"], ["class", "filter-content", 3, "selected", "click", 4, "ngFor", "ngForOf"], [1, "apply-btn"], [1, "app-btn", 3, "click"], [1, "side-menu-tabs", 3, "click"], [2, "margin-left", "22px"], ["type", "text", "matInput", "", 1, "filter-search-option", 3, "placeholder", "ngModel", "ngModelChange", "keyup"], [1, "filter-content", 3, "click"], ["class", "filter-data-list", 4, "ngIf"], [1, "filter-data-list"], [1, "time-input-wrapper"], ["matInput", "", "placeholder", "From Time", "readonly", "", 1, "filter-time-input", 3, "ngModel", "ngxTimepicker", "ngModelChange"], [3, "timeSet"], ["picker2", ""], ["matInput", "", "placeholder", "To Time", "readonly", "", 1, "filter-time-input", 2, "margin-left", "26px", 3, "ngModel", "ngxTimepicker", "ngModelChange"], ["picker3", ""]], template: function GenericListFilterComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵtemplate(2, GenericListFilterComponent_div_2_Template, 5, 5, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "span", 3);
        ɵngcc0.ɵɵlistener("click", function GenericListFilterComponent_Template_span_click_3_listener() { return ctx.openFilterSideBar(); });
        ɵngcc0.ɵɵelement(4, "i", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(5, GenericListFilterComponent_div_5_Template, 28, 7, "div", 5);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.filterKeys == null ? null : ctx.filterKeys.length);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", ctx.openFilterBar);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgClass, ɵngcc1.NgForOf, ɵngcc2.DefaultValueAccessor, ɵngcc2.NgControlStatus, ɵngcc2.NgModel, ɵngcc3.TimepickerDirective, ɵngcc3.NgxMaterialTimepickerComponent], pipes: [ɵngcc1.TitleCasePipe], styles: [".filter-modal-position-2[_ngcontent-%COMP%]{position:absolute;right:630px;top:109px}.filter-modal-position[_ngcontent-%COMP%]{position:absolute;right:474px;top:104px}.filter-modal-position-1[_ngcontent-%COMP%]{position:absolute;right:729px;top:109px}.modalbody[_ngcontent-%COMP%]{-moz-box-shadow:0 0 16px 0 rgba(0,0,0,.75);-webkit-box-shadow:0 0 16px 0 rgba(0,0,0,.75);background-color:#fff;border-radius:4px;box-shadow:0 0 16px 0 rgba(0,0,0,.75);display:flex;flex-direction:column;font-size:15px;height:400px;overflow:auto;position:fixed;width:400px;z-index:4}.ele[_ngcontent-%COMP%]{-webkit-animation:fadeIn .3s;-webkit-animation-fill-mode:forwards;animation:fadeIn .3s;animation-fill-mode:forwards;visibility:hidden}@-webkit-keyframes fadeIn{0%{opacity:0}to{opacity:1;visibility:visible}}@keyframes fadeIn{0%{opacity:0}to{opacity:1;visibility:visible}}.cancle-icon-1[_ngcontent-%COMP%]:hover{background-color:#f1f1f1;border-radius:50%;box-shadow:0 0 0 6px #f1f1f1}.header-item[_ngcontent-%COMP%]{display:flex;font-size:medium;justify-content:space-between;padding:10px 20px 0}.vl[_ngcontent-%COMP%]{border-left:1px solid grey;height:-webkit-fill-available;left:30%;margin-left:-4px;margin-top:38px;position:absolute;top:0}hr[_ngcontent-%COMP%]{margin-bottom:0}.side-menu-tabs[_ngcontent-%COMP%]{cursor:pointer;padding:14px 14px 14px 16px;width:116px}.side-menu-tabs[_ngcontent-%COMP%]:hover{background-color:#f1f1f1;color:#f28100}.side-bar-wrapper[_ngcontent-%COMP%]{width:117px}.main-body[_ngcontent-%COMP%]{display:flex}.filter-list-wrapper[_ngcontent-%COMP%]{width:100%}.filter-data-list[_ngcontent-%COMP%]{padding:10px}.filter-data-list-scroll-2[_ngcontent-%COMP%]{max-height:275px;overflow-y:scroll;padding:10px}.filter-content[_ngcontent-%COMP%]{cursor:pointer;padding:10px}.filter-content[_ngcontent-%COMP%]:hover{width:100%}.filter-content[_ngcontent-%COMP%]:hover, .filter-data-list[_ngcontent-%COMP%]   .filter-content.selected[_ngcontent-%COMP%]{background-color:#f1f1f1;color:#f28100}.cr-ptr[_ngcontent-%COMP%]{cursor:pointer}.cr-ptr[_ngcontent-%COMP%], .filter-side-menu[_ngcontent-%COMP%]   .side-menu-tabs.selected[_ngcontent-%COMP%]{color:#f28100}.reset-filter[_ngcontent-%COMP%]{color:#f28100;margin-right:10px}.reset-filter[_ngcontent-%COMP%]:hover{background-color:#f1f1f1;border-radius:0;box-shadow:0 0 0 6px #f1f1f1}.filter-time-input[_ngcontent-%COMP%]{background:transparent;border:none;border-bottom:1px solid #999;font-size:14px;height:30px!important;height:30px;margin-left:8px;outline:none;padding:13px 9px;width:150px!important}.time-input-wrapper[_ngcontent-%COMP%]{padding:10px 0}.icon-margin[_ngcontent-%COMP%]{margin-right:10px;margin-top:-1px}.apply-btn[_ngcontent-%COMP%]{bottom:10px;position:absolute;right:12px}.filter-search-option[_ngcontent-%COMP%]{border:none;border-bottom:1px solid #f28100;font-size:14px;margin-bottom:5px;margin-right:20px;outline:none;padding-bottom:5px;padding-top:8px;width:-webkit-fill-available}.filter-list[_ngcontent-%COMP%]{border:1px solid #f28100;border-radius:12px;color:#000;display:flex;font-size:10px;height:26px;margin:2px;padding:3px 8px}.filter-list-data[_ngcontent-%COMP%]{-webkit-box-orient:vertical;-webkit-line-clamp:1;display:-webkit-box;overflow:hidden;text-overflow:ellipsis;width:125px}.filter-list-container-2[_ngcontent-%COMP%]{color:#f28100;display:flex;margin-left:auto;margin-right:5px;margin-top:-1px}.flex-wrap[_ngcontent-%COMP%]{flex-wrap:wrap;justify-content:flex-end;max-width:385px}.cancle-icon[_ngcontent-%COMP%]{color:#f28100;cursor:pointer;font-size:14px;font-size:13px;font-weight:700;font-weight:500;padding-left:5px;position:relative;top:-1px}.cancle-icon[_ngcontent-%COMP%]   img[_ngcontent-%COMP%]{width:9px}.display-flex[_ngcontent-%COMP%]{display:flex}.filter-icon-position[_ngcontent-%COMP%]{color:#f28100;padding-right:15px}.filter-icon-position[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]{cursor:pointer}.filter-lst-show[_ngcontent-%COMP%]{display:flex;margin-left:auto}.filter-modal-position-3[_ngcontent-%COMP%]{position:absolute;right:569px;top:109px}.filter-color[_ngcontent-%COMP%]{color:#f28100;cursor:pointer}.dispatch-filters[_ngcontent-%COMP%]{position:relative}.dispatch-filters[_ngcontent-%COMP%]   .filter-position-modal[_ngcontent-%COMP%]{position:absolute;right:430px;top:24px}"] });
GenericListFilterComponent.ctorParameters = () => [];
GenericListFilterComponent.propDecorators = {
    list: [{ type: Input }],
    paramList: [{ type: Input }],
    onFilterChange: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GenericListFilterComponent, [{
        type: Component,
        args: [{
                selector: 'app-filter',
                template: "<div class=\"dispatch-filters display-flex\">\n    <div class=\"display-flex\">\n      <div *ngIf=\"filterKeys?.length\"\n        [ngClass]=\"(filterKeys.length>2)?'filter-list-container-2 flex-wrap':'filter-list-container-2'\">\n        <ng-container *ngIf=\"isCountClicked\">\n          <div class=\"filter-list\" *ngFor=\"let item of filterKeys;let i = index\">\n            <span class=\"filter-list-data\"> {{item | titlecase}} : {{allAppliedFilters[item]}} </span>\n            <span (click)=\"onRemoveFilter(item)\" class=\"cancle-icon\">\n              x\n            </span>\n          </div>\n        </ng-container>\n  \n  \n        <div class=\"filter-list\" *ngIf=\"!isCountClicked\">\n          <span  class=\"filter-list-data\"> {{filterKeys[0] | titlecase}} : {{allAppliedFilters[filterKeys[0]]}} </span>\n          <span (click)=\"onRemoveFilter(filterKeys[0])\" class=\"cancle-icon\">\n            x\n          </span>\n        </div>\n  \n        <div class=\"filter-list\" *ngIf=\"filterKeys?.length >= 2 && !isCountClicked\">\n          <span  class=\"filter-list-data\"> {{filterKeys[1] | titlecase}} : {{allAppliedFilters[filterKeys[1]]}} </span>\n          <span (click)=\"onRemoveFilter(filterKeys[1])\" class=\"cancle-icon\">\n            x\n          </span>\n        </div>\n  \n        <div (click)=\"openRestAppliedFilters()\" class=\"filter-list cr-ptr\"\n          *ngIf=\"filterKeys.length > 2 && !isCountClicked\">\n          +{{filterKeys.length - 2}}\n        </div>\n  \n      </div>\n      <span\n        class=\"filter-color\"\n        (click)=\"openFilterSideBar()\">\n        <i class=\"fa fa-filter fa-2x icon-margin\"></i>\n        <!-- <span *ngIf=\"totalFilterCount\">({{totalFilterCount}})</span>  -->\n      </span>\n    </div>\n  \n    <div *ngIf=\"openFilterBar\" class=\"filter-position-modal\">\n      <div class=\"modalbody ele\">\n        <div>\n          <div class=\"header-item\">\n            <div>\n              Filter\n            </div>\n            <div class=\"cr-ptr\">\n              <span (click)=\"resetAllFilters()\" class=\"reset-filter\">Reset</span>\n              <span (click)=\"onCancel()\"><img class=\"cancle-icon-1\" src=\"./../../assets/images/crossIcon.png\"></span>\n            </div>\n          </div>\n          <hr>\n  \n          <div class=\"main-body\">\n            <div class=\"side-bar-wrapper\">\n              <div class=\"filter-side-menu\">\n                <div \n                *ngFor=\"let filter of filterKeys1\"  \n                [class.selected]=\"selectedFilterTabType1 === filter || isFilterExists(filter)\"\n                (click)=\"selectedFilterTab(filter)\" class=\"side-menu-tabs\">\n                    {{filter | titlecase}}\n                </div>\n  \n                <div *ngIf=\"paramList?.fromTime && paramList?.toTime\"\n                [class.selected]=\"selectedFilterTabType1 === 'time' || timeFilter\" (click)=\"selectedFilterTab('time')\"\n                class=\"side-menu-tabs\">\n                Select Time\n               </div>\n              </div>\n            </div>\n  \n            <div class=\"vl\"></div>\n  \n            <div class=\"filter-list-wrapper\">\n              <div style=\"margin-left: 22px;\" *ngIf=\"selectedFilterTabType1 !== 'time'\">\n              <input class=\"filter-search-option\" type=\"text\" placeholder=\"Search {{selectedFilterTabType1}}\" matInput\n                [(ngModel)]=\"searchKey\" (keyup)=\"filterData(searchKey)\" />\n             </div>\n              <div  [ngClass]=\"(list[selectedFilterTabType1]?.length > 6) ? 'filter-data-list-scroll-2 filter-data-list' : 'filter-data-list'\">\n                <div \n                *ngFor=\"let item of list[selectedFilterTabType1]\"\n                class=\"filter-content\"\n                (click)=\"selectFilter(item,selectedFilterTabType1)\"\n                [class.selected]=\"selectedCategory.value === item\"\n                >\n                    {{item}}\n                </div>\n  \n                <div *ngIf=\"paramList.selectedTab > 0 && selectedFilterTabType1=='live status'\"\n                [ngClass]=\"(liveStatus?.length > 6) ? 'filter-data-list-scroll-2 filter-data-list' : 'filter-data-list'\">\n                <div (click)=\"selectFilter(item,'live status')\" [class.selected]=\"selectedCategory.value.value === item.value\"\n                  *ngFor=\"let item of liveStatus\" class=\"filter-content\">\n                  {{item.name}}\n                </div>\n                </div>\n  \n                <ng-container *ngIf=\"paramList?.fromTime && paramList?.toTime && selectedFilterTabType1 === 'time'\">\n                  <div *ngIf=\"selectedFilterTabType === 'time'\" class=\"filter-data-list\">\n                    <div class=\"time-input-wrapper\">\n                      <label>From Time</label>\n                      <input matInput class=\"filter-time-input\" [(ngModel)]=\"fromTime\" [ngxTimepicker]=\"picker2\"\n                        placeholder=\"From Time\" readonly>\n                      <ngx-material-timepicker (timeSet)=\"setTime('fromTime',$event)\" #picker2></ngx-material-timepicker>\n                    </div>\n    \n                    <div class=\"time-input-wrapper\">\n                      <label>To Time</label>\n                      <input style=\"margin-left: 26px;\" matInput class=\"filter-time-input\" [(ngModel)]=\"toTime\"\n                        [ngxTimepicker]=\"picker3\" placeholder=\"To Time\" readonly>\n                      <ngx-material-timepicker (timeSet)=\"setTime('toTime',$event)\" #picker3></ngx-material-timepicker>\n                    </div>\n                  </div>\n                </ng-container>\n  \n  \n              </div>\n             \n              <div class=\"apply-btn\">\n                <button (click)=\"applyFilters()\" class=\"app-btn\">\n                  <span>Apply</span>\n                </button>\n              </div>\n            </div>\n          </div>\n        </div>\n      </div>\n    </div>\n  </div>\n  ",
                styles: [".filter-modal-position-2{position:absolute;right:630px;top:109px}.filter-modal-position{position:absolute;right:474px;top:104px}.filter-modal-position-1{position:absolute;right:729px;top:109px}.modalbody{-moz-box-shadow:0 0 16px 0 rgba(0,0,0,.75);-webkit-box-shadow:0 0 16px 0 rgba(0,0,0,.75);background-color:#fff;border-radius:4px;box-shadow:0 0 16px 0 rgba(0,0,0,.75);display:flex;flex-direction:column;font-size:15px;height:400px;overflow:auto;position:fixed;width:400px;z-index:4}.ele{-webkit-animation:fadeIn .3s;-webkit-animation-fill-mode:forwards;animation:fadeIn .3s;animation-fill-mode:forwards;visibility:hidden}@-webkit-keyframes fadeIn{0%{opacity:0}to{opacity:1;visibility:visible}}@keyframes fadeIn{0%{opacity:0}to{opacity:1;visibility:visible}}.cancle-icon-1:hover{background-color:#f1f1f1;border-radius:50%;box-shadow:0 0 0 6px #f1f1f1}.header-item{display:flex;font-size:medium;justify-content:space-between;padding:10px 20px 0}.vl{border-left:1px solid grey;height:-webkit-fill-available;left:30%;margin-left:-4px;margin-top:38px;position:absolute;top:0}hr{margin-bottom:0}.side-menu-tabs{cursor:pointer;padding:14px 14px 14px 16px;width:116px}.side-menu-tabs:hover{background-color:#f1f1f1;color:#f28100}.side-bar-wrapper{width:117px}.main-body{display:flex}.filter-list-wrapper{width:100%}.filter-data-list{padding:10px}.filter-data-list-scroll-2{max-height:275px;overflow-y:scroll;padding:10px}.filter-content{cursor:pointer;padding:10px}.filter-content:hover{width:100%}.filter-content:hover,.filter-data-list .filter-content.selected{background-color:#f1f1f1;color:#f28100}.cr-ptr{cursor:pointer}.cr-ptr,.filter-side-menu .side-menu-tabs.selected{color:#f28100}.reset-filter{color:#f28100;margin-right:10px}.reset-filter:hover{background-color:#f1f1f1;border-radius:0;box-shadow:0 0 0 6px #f1f1f1}.filter-time-input{background:transparent;border:none;border-bottom:1px solid #999;font-size:14px;height:30px!important;height:30px;margin-left:8px;outline:none;padding:13px 9px;width:150px!important}.time-input-wrapper{padding:10px 0}.icon-margin{margin-right:10px;margin-top:-1px}.apply-btn{bottom:10px;position:absolute;right:12px}.filter-search-option{border:none;border-bottom:1px solid #f28100;font-size:14px;margin-bottom:5px;margin-right:20px;outline:none;padding-bottom:5px;padding-top:8px;width:-webkit-fill-available}.filter-list{border:1px solid #f28100;border-radius:12px;color:#000;display:flex;font-size:10px;height:26px;margin:2px;padding:3px 8px}.filter-list-data{-webkit-box-orient:vertical;-webkit-line-clamp:1;display:-webkit-box;overflow:hidden;text-overflow:ellipsis;width:125px}.filter-list-container-2{color:#f28100;display:flex;margin-left:auto;margin-right:5px;margin-top:-1px}.flex-wrap{flex-wrap:wrap;justify-content:flex-end;max-width:385px}.cancle-icon{color:#f28100;cursor:pointer;font-size:14px;font-size:13px;font-weight:700;font-weight:500;padding-left:5px;position:relative;top:-1px}.cancle-icon img{width:9px}.display-flex{display:flex}.filter-icon-position{color:#f28100;padding-right:15px}.filter-icon-position i{cursor:pointer}.filter-lst-show{display:flex;margin-left:auto}.filter-modal-position-3{position:absolute;right:569px;top:109px}.filter-color{color:#f28100;cursor:pointer}.dispatch-filters{position:relative}.dispatch-filters .filter-position-modal{position:absolute;right:430px;top:24px}"]
            }]
    }], function () { return []; }, { onFilterChange: [{
            type: Output
        }], list: [{
            type: Input
        }], paramList: [{
            type: Input
        }] }); })();

class GenericListFilterModule {
}
GenericListFilterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: GenericListFilterModule });
GenericListFilterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function GenericListFilterModule_Factory(t) { return new (t || GenericListFilterModule)(); }, imports: [[
            CommonModule,
            ReactiveFormsModule,
            FormsModule,
            NgxMaterialTimepickerModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(GenericListFilterModule, { declarations: function () { return [GenericListFilterComponent]; }, imports: function () { return [CommonModule,
        ReactiveFormsModule,
        FormsModule,
        NgxMaterialTimepickerModule]; }, exports: function () { return [GenericListFilterComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GenericListFilterModule, [{
        type: NgModule,
        args: [{
                declarations: [GenericListFilterComponent],
                imports: [
                    CommonModule,
                    ReactiveFormsModule,
                    FormsModule,
                    NgxMaterialTimepickerModule
                ],
                exports: [GenericListFilterComponent]
            }]
    }], null, null); })();

/*
 * Public API Surface of generic-list-filter
 */

/**
 * Generated bundle index. Do not edit.
 */

export { GenericListFilterComponent, GenericListFilterModule };
