// chalo-google-hybrid-login/esm5/lib/mobie-login/mobie-login.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Output, Input } from '@angular/core';
import { FormControl, Validators, FormBuilder } from '@angular/forms';
import { GoogleLoginService } from '../chalo-google-hybrid-login.service';
import { StatusCode } from '../models';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/forms';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/material/form-field';
import * as ɵngcc4 from '@angular/material/input';
import * as ɵngcc5 from '../directives/number.directive';
import * as ɵngcc6 from 'ng-otp-input';

function MobieLoginComponent_form_1_Template(rf, ctx) { if (rf & 1) {
    var _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "form", 2);
    ɵngcc0.ɵɵelementStart(1, "div");
    ɵngcc0.ɵɵelementStart(2, "mat-form-field", 3);
    ɵngcc0.ɵɵelement(3, "input", 4);
    ɵngcc0.ɵɵelementStart(4, "span", 5);
    ɵngcc0.ɵɵtext(5, "+91\u00A0");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "div", 6);
    ɵngcc0.ɵɵelementStart(7, "button", 7);
    ɵngcc0.ɵɵlistener("click", function MobieLoginComponent_form_1_Template_button_click_7_listener() { ɵngcc0.ɵɵrestoreView(_r3); var ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.sendOtp(true); });
    ɵngcc0.ɵɵelementStart(8, "a", 8);
    ɵngcc0.ɵɵtext(9, "Continue");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("formGroup", ctx_r0.sendOtpForm);
    ɵngcc0.ɵɵadvance(7);
    ɵngcc0.ɵɵproperty("disabled", !ctx_r0.sendOtpForm.valid);
} }
var _c0 = function () { return { length: 6, allowNumbersOnly: true, inputClass: "otpInput", containerClass: "otp-wrapper" }; };
function MobieLoginComponent_form_2_Template(rf, ctx) { if (rf & 1) {
    var _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "form", 2);
    ɵngcc0.ɵɵelementStart(1, "div", 9);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "div", 10);
    ɵngcc0.ɵɵelementStart(4, "ng-otp-input", 11);
    ɵngcc0.ɵɵlistener("onInputChange", function MobieLoginComponent_form_2_Template_ng_otp_input_onInputChange_4_listener($event) { ɵngcc0.ɵɵrestoreView(_r5); var ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.onOtpChange($event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "div", 6);
    ɵngcc0.ɵɵelementStart(6, "button", 7);
    ɵngcc0.ɵɵlistener("click", function MobieLoginComponent_form_2_Template_button_click_6_listener() { ɵngcc0.ɵɵrestoreView(_r5); var ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.verifyOtp(); });
    ɵngcc0.ɵɵelementStart(7, "a", 8);
    ɵngcc0.ɵɵtext(8, "Verify");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(9, "div", 12);
    ɵngcc0.ɵɵelementStart(10, "a", 13);
    ɵngcc0.ɵɵtext(11, "Did not recieve OTP? ");
    ɵngcc0.ɵɵelementStart(12, "span", 14);
    ɵngcc0.ɵɵlistener("click", function MobieLoginComponent_form_2_Template_span_click_12_listener() { ɵngcc0.ɵɵrestoreView(_r5); var ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.sendOtp(false); });
    ɵngcc0.ɵɵtext(13, "Resend OTP");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("formGroup", ctx_r1.verifyOtpForm);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1("Please enter the OTP sent to ", ctx_r1.number, "");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("config", ɵngcc0.ɵɵpureFunction0(4, _c0));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("disabled", !ctx_r1.verifyOtpForm.valid);
} }
var MobieLoginComponent = /** @class */ (function () {
    function MobieLoginComponent(fb, loginService) {
        this.fb = fb;
        this.loginService = loginService;
        this.onSignInResponse = new EventEmitter();
        this.showGoogleSignIn = new EventEmitter();
        this.showsendOtpForm = true;
        this.sendOtpForm = fb.group({
            number: new FormControl('', [Validators.required, Validators.minLength(10), Validators.maxLength(10)])
        });
        this.verifyOtpForm = fb.group({
            otp: new FormControl('', [Validators.required, Validators.minLength(6), Validators.maxLength(6)])
        });
    }
    /**
     * @return {?}
     */
    MobieLoginComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    MobieLoginComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (changes['showsendOtpForm']) {
            this.sendOtpForm.reset();
            this.verifyOtpForm.reset();
        }
    };
    /**
     * @param {?} hideSendOtpForm
     * @return {?}
     */
    MobieLoginComponent.prototype.sendOtp = /**
     * @param {?} hideSendOtpForm
     * @return {?}
     */
    function (hideSendOtpForm) {
        var _this = this;
        /** @type {?} */
        var selectedPhoneNo = this.sendOtpForm.get('number').value;
        this.number = selectedPhoneNo ? selectedPhoneNo : this.number;
        this.loginService.sendOtp(this.number).subscribe(function (res) {
            if (hideSendOtpForm) {
                _this.showsendOtpForm = false;
                // this.showGoogleSignIn.emit(false);
            }
        }, function (err) {
            _this.showsendOtpForm = true;
            // this.showGoogleSignIn.emit(true)
        });
    };
    /**
     * @return {?}
     */
    MobieLoginComponent.prototype.verifyOtp = /**
     * @return {?}
     */
    function () {
        var _this = this;
        /** @type {?} */
        var otp = this.verifyOtpForm.get('otp').value ? this.verifyOtpForm.get('otp').value : '';
        this.loginService.getMobileToken(otp, this.number).subscribe(function (data) {
            _this.onSignInResponse.emit({ code: StatusCode.Success, data: data });
        }, function (err) {
            _this.onSignInResponse.emit({ code: StatusCode.Failure, data: err });
        });
    };
    /**
     * @param {?} value
     * @return {?}
     */
    MobieLoginComponent.prototype.onOtpChange = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (value.length === 6) {
            this.verifyOtpForm.get('otp').setValue(value);
        }
        else {
            this.verifyOtpForm.get('otp').setValue('');
        }
        this.verifyOtpForm.get('otp').updateValueAndValidity();
    };
    /** @nocollapse */
    MobieLoginComponent.ctorParameters = function () { return [
        { type: FormBuilder },
        { type: GoogleLoginService }
    ]; };
    MobieLoginComponent.propDecorators = {
        onSignInResponse: [{ type: Output }],
        showGoogleSignIn: [{ type: Output }],
        showsendOtpForm: [{ type: Input }]
    };
MobieLoginComponent.ɵfac = function MobieLoginComponent_Factory(t) { return new (t || MobieLoginComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormBuilder), ɵngcc0.ɵɵdirectiveInject(GoogleLoginService)); };
MobieLoginComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MobieLoginComponent, selectors: [["mobie-login"]], inputs: { showsendOtpForm: "showsendOtpForm" }, outputs: { onSignInResponse: "onSignInResponse", showGoogleSignIn: "showGoogleSignIn" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 3, vars: 2, consts: [[1, "form-group"], [3, "formGroup", 4, "ngIf"], [3, "formGroup"], ["appearance", "outline"], ["matInput", "", "placeholder", "Enter mobile number", "formControlName", "number", "required", "", "autocomplete", "off", "numberDirective", "", "maxlength", "10"], ["matPrefix", ""], [1, "button-container"], ["type", "button", 1, "cstm-button", "active", 3, "disabled", "click"], [1, "btn-text"], [1, "sign-in-text"], [1, "otp-row"], [3, "config", "onInputChange"], [1, "anchor-div"], [1, "anchor-text"], [1, "resendText", 3, "click"]], template: function MobieLoginComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, MobieLoginComponent_form_1_Template, 10, 2, "form", 1);
        ɵngcc0.ɵɵtemplate(2, MobieLoginComponent_form_2_Template, 14, 5, "form", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showsendOtpForm);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.showsendOtpForm);
    } }, directives: [ɵngcc2.NgIf, ɵngcc1.ɵangular_packages_forms_forms_y, ɵngcc1.NgControlStatusGroup, ɵngcc1.FormGroupDirective, ɵngcc3.MatFormField, ɵngcc4.MatInput, ɵngcc1.DefaultValueAccessor, ɵngcc1.NgControlStatus, ɵngcc1.FormControlName, ɵngcc1.RequiredValidator, ɵngcc5.NumberDirective, ɵngcc1.MaxLengthValidator, ɵngcc3.MatPrefix, ɵngcc6.NgOtpInputComponent], styles: [".form-group[_ngcontent-%COMP%]{text-align:center;font-size:12px}.form-control[_ngcontent-%COMP%]{height:40px;border-radius:2px;box-shadow:none}.form-control[_ngcontent-%COMP%]:focus{border-color:#eee}.title[_ngcontent-%COMP%]{font-weight:600;font-size:16px}.anchor-div[_ngcontent-%COMP%]{margin-top:10px;font-size:14px}.resendText[_ngcontent-%COMP%]{color:#f47c21}.anchor-text[_ngcontent-%COMP%]{cursor:pointer;color:#000}.button-container[_ngcontent-%COMP%]{display:flex;justify-content:center}.cstm-button[_ngcontent-%COMP%]{background:0 0;border:1px solid #f47c21;border-radius:5px;text-align:center;width:100%;padding:8px 10px;display:inline-block;line-height:20px;margin-left:0!important}.btn-text[_ngcontent-%COMP%]{text-decoration:none;color:#f47c21;font-weight:700;font-size:14px}.cstm-button[_ngcontent-%COMP%]:disabled{cursor:not-allowed!important}.cstm-button.active[_ngcontent-%COMP%]{cursor:pointer;background:#f47c21}.cstm-button.active[_ngcontent-%COMP%]   .btn-text[_ngcontent-%COMP%]{color:#fff}.cstm-button[_ngcontent-%COMP%]:hover{cursor:pointer;background:#f47c21}.cstm-button[_ngcontent-%COMP%]:hover   .btn-text[_ngcontent-%COMP%]{color:#fff}.cstm-button[_ngcontent-%COMP%]:focus{outline:0}.phone-wrapper[_ngcontent-%COMP%]{display:flex;justify-content:space-around;align-items:baseline}.otp-row[_ngcontent-%COMP%]{display:flex;flex-direction:row;justify-content:center;width:100%;margin:10px auto 25px}  .mat-form-field-flex{border:0!important}  .mat-form-field-infix{padding:8px 16px}  .mat-form-field-infix:focus{outline:#f47c21 solid 2px}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MobieLoginComponent, [{
        type: Component,
        args: [{
                selector: 'mobie-login',
                template: "<div class=\"form-group\">\n  <form *ngIf=\"showsendOtpForm\"  [formGroup]=\"sendOtpForm\">\n    <div>\n     \n      <mat-form-field appearance=\"outline\">\n        <input matInput placeholder=\"Enter mobile number\" formControlName=\"number\"  required autocomplete=\"off\" numberDirective maxlength=\"10\">\n        <span matPrefix>+91&nbsp;</span>\n      </mat-form-field>\n      <div class=\"button-container\">\n        <button class=\"cstm-button active\" type=\"button\" (click)=\"sendOtp(true)\" [disabled]=\"!sendOtpForm.valid\"><a class=\"btn-text\">Continue</a>\n        </button>\n      </div>\n    </div>\n  </form>\n  <form  *ngIf=\"!showsendOtpForm\" [formGroup]=\"verifyOtpForm\">\n    <div class=\"sign-in-text\">Please enter the OTP sent to {{number}}</div>\n    <div class=\"otp-row\">\n      <ng-otp-input  (onInputChange)=\"onOtpChange($event)\"  [config]=\"{length:6,allowNumbersOnly:true,inputClass:'otpInput',containerClass:'otp-wrapper'}\"></ng-otp-input>\n    </div>\n    <div class=\"button-container\">\n      <button class=\"cstm-button active\" type=\"button\" (click)=\"verifyOtp()\" [disabled]=\"!verifyOtpForm.valid\"><a class=\"btn-text\">Verify</a>\n      </button>\n    </div>\n    <div class=\"anchor-div\"><a class=\"anchor-text\" >Did not recieve OTP? <span class=\"resendText\" (click)=\"sendOtp(false)\">Resend OTP</span></a></div>\n\n  </form>\n</div>\n",
                styles: [".form-group{text-align:center;font-size:12px}.form-control{height:40px;border-radius:2px;box-shadow:none}.form-control:focus{border-color:#eee}.title{font-weight:600;font-size:16px}.anchor-div{margin-top:10px;font-size:14px}.resendText{color:#f47c21}.anchor-text{cursor:pointer;color:#000}.button-container{display:flex;justify-content:center}.cstm-button{background:0 0;border:1px solid #f47c21;border-radius:5px;text-align:center;width:100%;padding:8px 10px;display:inline-block;line-height:20px;margin-left:0!important}.btn-text{text-decoration:none;color:#f47c21;font-weight:700;font-size:14px}.cstm-button:disabled{cursor:not-allowed!important}.cstm-button.active{cursor:pointer;background:#f47c21}.cstm-button.active .btn-text{color:#fff}.cstm-button:hover{cursor:pointer;background:#f47c21}.cstm-button:hover .btn-text{color:#fff}.cstm-button:focus{outline:0}.phone-wrapper{display:flex;justify-content:space-around;align-items:baseline}.otp-row{display:flex;flex-direction:row;justify-content:center;width:100%;margin:10px auto 25px}::ng-deep .mat-form-field-flex{border:0!important}::ng-deep .mat-form-field-infix{padding:8px 16px}::ng-deep .mat-form-field-infix:focus{outline:#f47c21 solid 2px}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormBuilder }, { type: GoogleLoginService }]; }, { onSignInResponse: [{
            type: Output
        }], showGoogleSignIn: [{
            type: Output
        }], showsendOtpForm: [{
            type: Input
        }] }); })();
    return MobieLoginComponent;
}());
export { MobieLoginComponent };
if (false) {
    /** @type {?} */
    MobieLoginComponent.prototype.onSignInResponse;
    /** @type {?} */
    MobieLoginComponent.prototype.showGoogleSignIn;
    /** @type {?} */
    MobieLoginComponent.prototype.showsendOtpForm;
    /** @type {?} */
    MobieLoginComponent.prototype.sendOtpForm;
    /** @type {?} */
    MobieLoginComponent.prototype.verifyOtpForm;
    /** @type {?} */
    MobieLoginComponent.prototype.showVerifyOtpForm;
    /** @type {?} */
    MobieLoginComponent.prototype.number;
    /** @type {?} */
    MobieLoginComponent.prototype.fb;
    /** @type {?} */
    MobieLoginComponent.prototype.loginService;
}
